volmdlr/wires.py

```diff
@@ -904,66 +903,67 @@ def shared_primitives_extremities(self, contour):
-
-
-            elif edge2.point_belongs(edge1.end):
-                # list_p.add(edge1.end)
-                # instead of point not in list_p (due to errors)
-                if list_p == []:
-                    list_p.append(edge1.end)
-                if list_p != [] and edge1.end.point_distance(edge1.end.nearest_point(list_p)) > 1e-4:
-                    list_p.append(edge1.end)
+            edges1 = set()
+            # edges2 = set()
+            list_p = []
+    
+            for edge1, edge2 in itertools.product(self.primitives,
+                                                  contour.primitives):
+                if edge1.point_belongs(edge2.start, 1e-5):
+                    # list_p.add(edge2.start)
+                    # instead of point not in list_p (due to errors)
+                    if list_p == []:
+                        list_p.append(edge2.start)
+                    if list_p != [] and edge2.start.point_distance(edge2.start.nearest_point(list_p)) > 1e-4:
+                        list_p.append(edge2.start)
+    
+                    # edges2.add(edge2)
+                    edges1.add(edge1)
+    
+                elif edge2.point_belongs(edge1.start, 1e-5):
+                    # list_p.add(edge1.start)
+                    # instead of point not in list_p (due to errors)
+                    if list_p == []:
+                        list_p.append(edge1.start)
+                    if list_p != [] and edge1.start.point_distance(edge1.start.nearest_point(list_p)) > 1e-4:
+                        list_p.append(edge1.start)
+    
+                    edges1.add(edge1)
+                    # edges2.add(edge2)
+    
+                elif edge1.point_belongs(edge2.end, 1e-5):
+                    # list_p.add(edge2.end)
+                    # instead of point not in list_p (due to errors)
+                    if list_p == []:
+                        list_p.append(edge2.end)
+                    if list_p != [] and edge2.end.point_distance(edge2.end.nearest_point(list_p)) > 1e-4:
+                        list_p.append(edge2.end)
+    
+                    # edges2.add(edge2)
+                    edges1.add(edge1)
+    
+                elif edge2.point_belongs(edge1.end, 1e-5):
+                    # list_p.add(edge1.end)
+                    # instead of point not in list_p (due to errors)
+                    if list_p == []:
+                        list_p.append(edge1.end)
+                    if list_p != [] and edge1.end.point_distance(edge1.end.nearest_point(list_p)) > 1e-4:
+                        list_p.append(edge1.end)
+    
+                    edges1.add(edge1)
+                    # edges2.add(edge2)
+    
+            edges1 = list(edges1)
+            # edges2 = list(edges2)
 
-                edges1.add(edge1)
-                # edges2.add(edge2)
+            return edges1, list_p
 
-        edges1 = list(edges1)
-        # edges2 = list(edges2)
+        def order_extremities_points(self, contour):
```

Comment:
Same goes for here.

Reply:
I tried to refactor this method differently, but, it gives **C** as complexity index.